slicedimage/io.py

```diff
@@ -27,8 +27,15 @@ def infer_backend(baseurl, allow_caching=True):
 
     if parsed.scheme in ("http", "https"):
         backend = HttpBackend(baseurl)
-        if allow_caching:
+        if allow_caching is True:
```

Comment:
Proposal: let's just make this a breaking change and change this to:

```
def infer_backend(baseurl, cache_config=None):
    cache_config = cache_config or {}
    if cache_config.get('allow_caching', False) and 'directory' in cache_config:
        # set up cache
```

Reply:
If that's ok with you, :+1: I'd propose `backend_config` though which might mean that at the starfish level it should become:

```
{
  'backend': { # or 'slicedimage' for when/if other backends are added?
    'cache':
      {
      }
  }
}
```

Reply:
Done using a "backend" sub-dictionary everywhere.